pkg/generators/backend/generator.go

```diff
@@ -92,94 +97,210 @@ func NewGenerator(instance, namespace string, spec saasv1alpha1.BackendSpec) Gen
+		fmt.Sprintf("%s/traffic", saasv1alpha1.GroupVersion.Group): fmt.Sprintf("%s-%s", component, listener),
+	}
 }
 
 // WorkerGenerator has methods to generate resources for a
 // Backend environment
 type WorkerGenerator struct {
-	generators.BaseOptions
+	generators.BaseOptionsV2
 	Image      saasv1alpha1.ImageSpec
 	WorkerSpec saasv1alpha1.WorkerSpec
 	Options    config.WorkerOptions
 }
 
-// HPA returns a basereconciler.GeneratorFunction
-func (gen *WorkerGenerator) HPA() basereconciler.GeneratorFunction {
-	key := types.NamespacedName{Name: gen.Component, Namespace: gen.Namespace}
-	return hpa.New(key, gen.GetLabels(), *gen.WorkerSpec.HPA)
-}
+// Validate that WorkerGenerator implements workloads.DeploymentWorkload interface
+var _ workloads.DeploymentWorkload = &WorkerGenerator{}
 
-// PDB returns a basereconciler.GeneratorFunction
-func (gen *WorkerGenerator) PDB() basereconciler.GeneratorFunction {
-	key := types.NamespacedName{Name: gen.Component, Namespace: gen.Namespace}
-	return pdb.New(key, gen.GetLabels(), gen.Selector().MatchLabels, *gen.WorkerSpec.PDB)
+func (gen *WorkerGenerator) Deployment() basereconciler_resources.DeploymentTemplate {
+	return basereconciler_resources.DeploymentTemplate{
+		Template: gen.deployment(),
+		RolloutTriggers: []basereconciler_resources.RolloutTrigger{
+			{Name: "backend-system-events-hook", SecretName: pointer.String("backend-system-events-hook")},
+			{Name: "backend-error-monitoring", SecretName: pointer.String("backend-error-monitoring")},
+		},
+		EnforceReplicas: gen.WorkerSpec.HPA.IsDeactivated(),
+		IsEnabled:       true,
+	}
 }
-
-// PodMonitor returns a basereconciler.GeneratorFunction
-func (gen *WorkerGenerator) PodMonitor() basereconciler.GeneratorFunction {
-	key := types.NamespacedName{Name: gen.Component, Namespace: gen.Namespace}
-	return podmonitor.New(key, gen.GetLabels(), gen.Selector().MatchLabels,
+func (gen *WorkerGenerator) HPASpec() *saasv1alpha1.HorizontalPodAutoscalerSpec {
+	return gen.WorkerSpec.HPA
+}
+func (gen *WorkerGenerator) PDBSpec() *saasv1alpha1.PodDisruptionBudgetSpec {
+	return gen.WorkerSpec.PDB
+}
+func (gen *WorkerGenerator) MonitoredEndpoints() []monitoringv1.PodMetricsEndpoint {
+	return []monitoringv1.PodMetricsEndpoint{
 		podmonitor.PodMetricsEndpoint("/metrics", "metrics", 30),
-	)
+	}
 }
 
 // CronGenerator has methods to generate resources for a
 // Backend environment
 type CronGenerator struct {
-	generators.BaseOptions
+	generators.BaseOptionsV2
 	Image    saasv1alpha1.ImageSpec
 	CronSpec saasv1alpha1.CronSpec
 	Options  config.CronOptions
 }
+
+// Validate that CronGenerator implements workloads.DeploymentWorkload interface
+var _ workloads.DeploymentWorkload = &CronGenerator{}
+
+func (gen *CronGenerator) Deployment() basereconciler_resources.DeploymentTemplate {
+	return basereconciler_resources.DeploymentTemplate{
+		Template:        gen.deployment(),
+		RolloutTriggers: nil,
```

Comment:
backend-cron mounts secret `backend-error-monitoring`, should it have the rollout triggers enabled for that secret?

Reply:
Yep, that's an error. Fixed with 8e9300a